addons/event_sale/models/sale_order.py

```diff
@@ -169,13 +163,17 @@ def _get_sale_order_line_multiline_description_sale(self):
 
     def _get_display_price(self):
         if self.event_ticket_id and self.event_id:
-            event_ticket = self.event_ticket_id.with_context(
-                pricelist=self.order_id.pricelist_id.id,
-                uom=self.product_uom.id
-            )
-            if self.order_id.pricelist_id.discount_policy == 'with_discount':
-                return event_ticket.price_reduce
+            event_ticket = self.event_ticket_id
+            company = event_ticket.company_id or self.env.company
+            currency = company.currency_id
+            product = event_ticket.product_id
+            pricelist = product.product_tmpl_id._get_contextual_pricelist()
+            if pricelist.discount_policy == "with_discount":
+                price = event_ticket.price_reduce
```

Comment:
Why do you remove the contextual information?
I think it's necessary to provide the right information in the context (pricelist, uom, qty) to get the correct `price_reduce` :thinking:

Reply:
Thanks!